libs/web3/src/lib/web3-connectors.ts

```diff
@@ -118,7 +114,31 @@ export const createConnectors = (
     initializeUrlConnector(localProviderUrl, walletMnemonic),
     initializeMetaMaskConnector(),
     initializeCoinbaseConnector(providerUrl),
-    initializeWalletConnector(WALLETCONNECT_PROJECT_ID, chainId, providerUrl),
-    initializeWalletConnectLegacyConnector(chainId, providerUrl),
+    initializeWalletConnector(WALLETCONNECT_PROJECT_ID, {
+      [chainId]: providerUrl,
+    }),
+    initializeWalletConnectLegacyConnector({
+      [chainId]: providerUrl,
+    }),
+  ].filter(Boolean) as unknown as [Connector, Web3ReactHooks][];
+};
+
+export const createMultiChainConnectors = (
+  transports: { [chainId: number]: string },
+  localProviderUrl?: string,
+  walletMnemonic?: string
+) => {
+  if (Object.values(transports).length === 0) {
+    throw new Error('ERR_CONNECTORS: could not create connectors');
+  }
+
+  return [
+    initializeUrlConnector(localProviderUrl, walletMnemonic),
+    initializeMetaMaskConnector(),
+    // TODO: This needs to be tested out as it only allows single rpc connection
```

Comment:
@daro-maj can you check this

Reply:
Tested it with coinbase and it works fine for me